src/librustc_span/lib.rs

```diff
@@ -1075,7 +1116,10 @@ impl SourceFile {
         unmapped_path: FileName,
         mut src: String,
         start_pos: BytePos,
+        checksum_kind: Option<SourceFileChecksumKind>,
     ) -> Self {
+        // Compute the file checksum *before* any normalization.
+        let src_file_checksum = checksum_kind.map(|k| SourceFileChecksum::new(k, src.as_bytes()));
```

Comment:
We should probably do the same with `src_hash` then? Ideally they get computed all at once.